scala-package/infer/src/main/scala/org/apache/mxnet/infer/ObjectDetector.scala

```diff
@@ -147,13 +150,16 @@ class ObjectDetector(modelPathPrefix: String,
   def imageBatchObjectDetect(inputBatch: Traversable[BufferedImage], topK: Option[Int] = None):
   IndexedSeq[IndexedSeq[(String, Array[Float])]] = {
 
-    val imageBatch = ListBuffer[NDArray]()
-    for (image <- inputBatch) {
-      val scaledImage = ImageClassifier.reshapeImage(image, width, height)
-      val pixelsNdarray = ImageClassifier.bufferedImageToPixels(scaledImage, inputShape)
-      imageBatch += pixelsNdarray
-    }
-    val op = NDArray.concatenate(imageBatch)
+    val inputBatchSeq = inputBatch.toIndexedSeq
+    val imageBatch = inputBatchSeq.indices.par.map(idx => {
+      val scaledImage = ImageClassifier.reshapeImage(inputBatchSeq(idx), width, height)
```

Comment:
we should parallelize this method using threads - you can create a JIRA for it and pick up whenever time permits

Reply:
`.par` already done the parallelism.

Reply:
oh cool, does it spawn threads?

Reply:
`par` will spawn as many threads as the default `ForkJoinThreadPool` size if there is no other concurrent process.